app/src/main/java/com/example/hackernews/viewmodel/MainViewModel.kt

```diff
@@ -43,25 +65,102 @@ class MainViewModel(private val newsRepository: NewsRepository) : ViewModel() {
     }
 
     private fun fetchNews(type: NewsDataType = NewsDataType.TOP_STORIES) {
-        newsRepository.getNews(type) { news, error ->
+        Log.d("FETCHING()", type.rawValue)
+        newsRepository.getNews(type) { singleNews, error ->
             if (error != null) {
                 handleError(error)
                 return@getNews
             }
+            attachNewsToLiveData(selectedNewsType, singleNews!!)
+        }
+    }
+
+    fun attachNewsToLiveData(type: NewsDataType, news: News) {
```

Comment:
The method name is a bit off 😄 `handleNews` would be better IMHO. 
Anyway, using this method might give the wrong UI. If for example, the user selects saved news, newDataType is changed to saved news, and a call made via repository for saved news. If now user changes his mind and selects top news before you get back saved news from the repository, then newDataType gets changed to top stories and repository is asked for top stories which takes a while and when saved news are returned they'll be shown as top stories. 😄